website/docs/BestPractices.md

```diff
@@ -0,0 +1,203 @@
+---
+id: bestpractices
+title: Best Practices
+---
+
+# Best Practices
+
+This guide aims to share our best practices that help you write performant and resilient tests. 
+
+## Use resilient selectors
+
+__Note:__ To find out all the possible selectors WebdriverIO supports, checkout our [Selectors](./Selectors.md) page.
+
+Use selectors that are resilient to changes in the DOM.
+
+```js reference useResilientSelectors
+await $('button[type="submit"]');
+```
+
+## Limit DOM interactions
+
+Everytime you use the [`$`](https://webdriver.io/docs/api/browser/$) or [`$$`](https://webdriver.io/docs/api/browser/$$) command, WebdriverIO tries to locate the element in the DOM. These operations are very expensive so you should try to limit them as much as possible.
+
+### Prefer using a single selector over chaining
+
+When using the [`$`](https://webdriver.io/docs/api/browser/$) or [`$$`](https://webdriver.io/docs/api/browser/$$) command, the only time you should use chaining is when you want to combine different [selector strategies](https://webdriver.io/docs/selectors/#custom-selector-strategies).
+
+Using a single strategy
+
+```js
+// 👎
+await $('table').$('tr').$('td');
+```
+
+``` js
+// 👍
+await $('table').custom$('myCustomStrategy', 'tr').$('td');
+```
+
+Another reason why you do not want to chain elements and would want to use a single selector is because it locates the elements separately and since DOM interactions are expensive, this should be avoided.
```

Comment:
> DOM interactions are expensive

I think this is not correct. We don't interact with the DOM, we just fetch elements. The expensive part is making the request.